agent/hcp/telemetry/otel_sink.go

```diff
@@ -85,28 +89,19 @@ func NewOTELSink(opts *OTELSinkOpts) (*OTELSink, error) {
 		return nil, fmt.Errorf("ferror: provide valid context")
 	}
 
-	logger := hclog.FromContext(opts.Ctx).Named("otel_sink")
-
-	filterList, err := newFilterRegex(opts.Filters)
-	if err != nil {
-		logger.Error("Failed to initialize all filters", "error", err)
+	if opts.ConfigProvider == nil {
+		return nil, fmt.Errorf("ferror: provide valid config provider")
 	}
 
-	attrs := make([]attribute.KeyValue, 0, len(opts.Labels))
```

Comment:
Don't need labels / filter logic in here since the provider will give this information.